mcs/class/monodoc/Test/Monodoc.Ecma/EcmaUrlTests.cs

```diff
@@ -454,38 +470,293 @@ public void ExplicitIndexerImplementation ()
+				Namespace = "System.IO",
+				TypeName = "Path",
+				MemberName = "TheField"
+			};
+
+			AssertEcmaString ("F:System.IO.Path.TheField", actual);
+		}
+
+		[Test]
+		public void ToEcmaCref_ExplicitlyImplemented_Field ()
+		{
+			var explicitImpl = new EcmaDesc {
+				Namespace = "System.Web.SessionState",
+				TypeName = "IHttpSessionState",
+				MemberName = "Item",
+			};
+			var actual = new EcmaDesc {
+				DescKind = EcmaDesc.Kind.Field,
+				TypeName = "HttpSessionStateContainer",
+				Namespace = "System.Web.SessionState",
+				ExplicitImplMember = explicitImpl,
+			};
+			AssertEcmaString ("F:System.Web.SessionState.HttpSessionStateContainer$System.Web.SessionState.IHttpSessionState.Item", actual);
+		}		
+
+		[Test]
+		public void ToEcmaCref_Property ()
+		{
+			var actual = new EcmaDesc {
+				DescKind = EcmaDesc.Kind.Property,
+				Namespace = "System.IO",
+				TypeName = "Path",
+				MemberName = "TheProperty"
+			};
+
+			AssertEcmaString ("P:System.IO.Path.TheProperty", actual);
+		}
+
+		[Test]
+		public void ToEcmaCref_ExplicitlyImplemented_Property ()
+		{
+			var explicitImpl = new EcmaDesc {
+				Namespace = "System.Web.SessionState",
+				TypeName = "IHttpSessionState",
+				MemberName = "Item"
+			};
+			var actual = new EcmaDesc {
+				DescKind = EcmaDesc.Kind.Property,
+				TypeName = "HttpSessionStateContainer",
+				Namespace = "System.Web.SessionState",
+				ExplicitImplMember = explicitImpl,
+			};
+			AssertEcmaString ("P:System.Web.SessionState.HttpSessionStateContainer$System.Web.SessionState.IHttpSessionState.Item", actual);
 		}
 
-		bool IsEcmaNode (Node node)
+		[Test]
+		public void ToEcmaCref_ExplicitlyImplemented_Method ()
 		{
-			var url = node.PublicUrl;
-			return url != null && url.Length > 2 && url[1] == ':';
-		}*/
+			var explicitImpl = new EcmaDesc {
+				Namespace = "System.Web.SessionState",
+				TypeName = "IHttpSessionState",
+				MemberName = "Item",
+				MemberArguments = new [] { new EcmaDesc { DescKind = EcmaDesc.Kind.Type, Namespace = "System", TypeName = "Int32" } },
+			};
+			var actual = new EcmaDesc {
+				DescKind = EcmaDesc.Kind.Method,
+				TypeName = "HttpSessionStateContainer",
+				Namespace = "System.Web.SessionState",
+				ExplicitImplMember = explicitImpl,
+			};
+			AssertEcmaString ("M:System.Web.SessionState.HttpSessionStateContainer$System.Web.SessionState.IHttpSessionState.Item(System.Int32)", actual);
+		}		
+
+		[Test]
+		public void ToEcmaCref_Event ()
+		{
+			var actual = new EcmaDesc {
+				DescKind = EcmaDesc.Kind.Event,
+				Namespace = "System.IO",
+				TypeName = "Path",
+				MemberName = "TheEvent"
+			};
+
+			AssertEcmaString ("E:System.IO.Path.TheEvent", actual);
+		}
+
+		[Test]
+		public void ToEcmaCref_Operator ()
+		{
+			var actual = new EcmaDesc {
+				DescKind = EcmaDesc.Kind.Operator,
+				Namespace = "System",
+				TypeName = "Int32",
+				MemberName = "Addition"
+			};
+
+			AssertEcmaString ("O:System.Int32.Addition", actual);
```

Comment:
While interesting, AFAIK, `O:` isn't a known/existing ECMA prefix.

At least, it's not in [ECMA-334 4th Edition](http://www.ecma-international.org/publications/files/ECMA-ST/Ecma-334.pdf) §E.3.1 **ID string format**.

Is there any reason you can't use `M:` with the IL names, e.g. `M:System.Int32.op_Addition`?